src/main/java/com/example/onlinebookstore/validation/impl/FieldMatchValidator.java

```diff
@@ -0,0 +1,33 @@
+package com.example.onlinebookstore.validation.impl;
+
+import com.example.onlinebookstore.validation.FieldMatch;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String field;
+    private String fieldMatch;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.field = constraintAnnotation.field();
+        this.fieldMatch = constraintAnnotation.fieldMatch();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        try {
+            var fieldValue = value.getClass().getDeclaredField(field);
+            var fieldMatchValue = value.getClass().getDeclaredField(fieldMatch);
+            fieldValue.setAccessible(true);
+            fieldMatchValue.setAccessible(true);
+
+            Object fieldVal = fieldValue.get(value);
+            Object fieldMatchVal = fieldMatchValue.get(value);
```

Comment:
Code duplication, logic of getting field could be moved to method

Reply:
Done